src/pages/Generate.tsx

```diff
@@ -1,7 +1,33 @@
-import React from "react";
-import { Link } from "react-router-dom";
-import { Button, Flex } from "@chakra-ui/react";
+import React, { useState } from "react";
+import {
+  Button,
+  Container,
+  Flex,
+  FormLabel,
+  Heading,
+  Input,
+  Select,
+} from "@chakra-ui/react";
+import { gql, useMutation } from "@apollo/client";
+
+const GENERATE = gql`
+  mutation GenerateSchedule($year: String!, $term: String!) {
+    generateSchedule(year: $year, term: $term) {
+      year
+      term
+    }
+  }
+`;
+
 export const Generate = () => {
+  const [submit, { data, loading, error }] = useMutation(GENERATE);
+  const [year, setYear] = useState("");
+  const [term, setTerm] = useState("");
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    submit({ variables: { year, term } });
+    e.preventDefault();
+  };
```

Comment:
We probably want this to redirect to the view schedules page if the request succeeds right?

Reply:
Yeah, I was just leaving it until the backend exists so I can check the success properly from their response